_posts/2021-02-19-26-rln-relay.md

```diff
@@ -0,0 +1,168 @@
+
+
+#### Economic-Incentive Spam protection
+But, is it the end of the spam-protection journey!? shall we get along with the spammers! Certainly not, Waku is here to save you!
+We brought you a p2p spam-protection method which:
+- is **efficient**  and fits the  network of **heterogeneous** peers (i.e., no crazy computational, storage, memory, and bandwidth requirement!). 
+- respects users privacy (unlike reputation-based and centralized methods)
+- deploys economic-incentives to contain spammers' activity. Namely, there is a financial sacrifice for those who want to spam the system. How? follow along ...
+
+We devise a general rule to save everyone's life and that is 
+
+**No one can publish more than M messages per epoch without being financially charged!** 
+
+We set M to 1 for now, but keep in mind, it can be any arbitrary value.
+You may be thinking that it is too restrictive! one per epoch!? but don't worry, we set the epoch to a reasonable value so that it does not slow down the communication of innocent users but will make the life of spammers harder! Epoch here can be every second, as defined by UTC date-time +-20s.
+
+
+The remainder of this blog is all about the story of how to enforce this limit on each user's messaging rate as well as how to impose the financial cost when the limit gets violated.  This brings us to the Rate Limiting Nullifiers and how we integrate this technique into Waku v2 (in specific the `WakuRelay` protocol) to protect our valuable users against spammers.
+
+
+# Technical Terms
+
+**Zero-knowledge proof**: Zero-knowledge proof (ZKP) allows a *prover* to show a *verifier* that they know something, without revealing what that something is. This means you can do trust-minimized computation that is also privacy preserving. As a basic example, instead of showing your ID when going to a bar you simply give them a proof that you are over 18, without showing the doorman your id. In this write up, by ZKP we essentially mean zkSNARK which is one of the many types of ZKPs. 
+
+**Threshold Secret Sharing Scheme**: (m,n) Threshold secret-sharing is a method by which you can split a secret value s into let's say n pieces in a way that the secret s can be reconstructed by having m pieces (m < n). The economic-incentive spam protection utilizes a (2,n) secret sharing scheme instantiated by Shamir Secret Sharing Scheme.
+
+# Overview: Economic-Incentive Spam protection through Rate Limiting Nullifiers
+
+
+The following are the foundational components of Economic-incentive spam protection:
```

Comment:
This seems implementation specific. I'd phrase it as - the fundamental point is to make it economically costly to send more than your share of messages, and to do so in a privacy-preserving fashion. To do that we have the following components ~